template/typescript/package.json

```diff
@@ -7,15 +7,19 @@
   "repository": "{{repo}}",
   "main": "dist/index.js",
   "module": "dist/index.es.js",
+  "types": "dist/index.d.ts",
   "jsnext:main": "dist/index.es.js",
+  "files": [
+    "dist"
+  ],
   "engines": {
     "node": ">=8",
     "npm": ">=5"
   },
   "scripts": {
     "test": "cross-env CI=1 react-scripts test --env=jsdom",
     "test:watch": "react-scripts test --env=jsdom",
-    "build": "rollup -c",
+    "build": "rollup -c && tsc -d --emitDeclarationOnly --noEmit false --declarationDir dist",
```

Comment:
A workaround for `@rollup/plugin-typescript`, see https://github.com/rollup/plugins/issues/287#issuecomment-650581821